pkg/kotsadmupstream/upstream.go

```diff
@@ -66,6 +67,22 @@ func DownloadUpdate(appID string, update types.Update, skipPreflights bool, skip
 		}
 
 		errMsg := finalError.Error()
+
+		var kotsApplication *kotsv1beta1.Application
```

Comment:
i don't think this block of code is needed at all? the pull function will already return the incompatibility message as an actionable error, and will know if it's an install or not. you probably added this before rebasing?